pkg/controller/controllerutil/controllerutil.go

```diff
@@ -324,12 +322,17 @@ func CreateOrPatch(ctx context.Context, c client.Client, obj client.Object, f Mu
 
 // mutate wraps a MutateFn and applies validation to its result
 func mutate(f MutateFn, key client.ObjectKey, obj client.Object) error {
-	if err := f(); err != nil {
-		return err
-	}
-	if newKey := client.ObjectKeyFromObject(obj); key != newKey {
-		return fmt.Errorf("MutateFn cannot mutate object name and/or object namespace")
+	// if f is nil, there is no mutate operation to be done, in which case just return
+	if f != nil {
+
```

Comment:
```suggestion
	// if f is nil, there is no mutate operation to be done, in which case just return
	if f == nil {
	  return nil
	}
```

Prefer to return right away rather than having everything in an if block

Reply:
@vincepri happy to do this. The trade-off here is now you have 2 locations (top and bottom) of func where a successful return can occur.

Just want to check you prefer that relative to the `if` as it stands.

Let me know.

Reply:
+1 on the @vincepri suggestion. having multiple if-else creates branching. Should just check the terminating condition first